### PR TITLE
feat(storybook): use uni-builder instead of modern.js builder as build tool

### PR DESCRIPTION
**Pull request by 9aoy**
## Summary

modern.js builder is no longer maintained, should use uni-builder or rsbuild instead.

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/storybook/builder/src/core.ts [hunk 1]
```diff
@@ -27,27 +28,24 @@ export async function getCompiler(
...
-    }
-  }
-
-  const builder = await createBuilder(provider, {
+  const builder = await createUniBuilder({
+    bundlerType,
     cwd,
     target: 'web',
-    framework: 'modern.js storybook',
+    frameworkConfigPath: res?.path ? res?.path : undefined,
```
Comment (chenjiahan) [inline packages/storybook/builder/src/core.ts:44]:
```suggestion
    frameworkConfigPath: res?.path ? res.path : undefined,
```

File: packages/storybook/builder/package.json [hunk 2]
```diff
@@ -56,23 +56,24 @@
   },
   "license": "MIT",
   "dependencies": {
-    "@modern-js/builder": "workspace:*",
-    "@modern-js/builder-shared": "workspace:*",
+    "@rsbuild/core": "0.4.3",
```
Comment (chenjiahan) [inline packages/storybook/builder/package.json:59]: is @rsbuild/core used?

File: packages/storybook/builder/src/types.ts [hunk 3]
```diff
@@ -1,21 +1,18 @@
-import type { BuilderConfig as WebpackBuilderConfig } from '@modern-js/builder-webpack-provider';
-import type { BuilderConfig as RspackBuilderConfig } from '@modern-js/builder-rspack-provider';
-import { BuilderPlugin } from '@modern-js/builder-shared';
+import type { RsbuildPlugin } from '@rsbuild/shared';
```
Comment (chenjiahan) [inline packages/storybook/builder/src/types.ts:1]: It is recommended to import types / methods from `@rsbuild/core`, as `@rsbuild/shared` is a private package of Rsbuild and may change in non-major versions.

File: packages/storybook/builder/package.json (hunk 2 above)
Comment (9aoy) [reply packages/storybook/builder/package.json]: removed

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**